src/DurableTask.SqlServer/SqlConnectionFactory.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading.Tasks;
+
+using Azure.Core;
+using Azure.Identity;
+
+using Microsoft.Data.SqlClient;
+
+namespace DurableTask.SqlServer
+{
+    internal class SqlConnectionFactory
+    {
+        readonly string connectionString;
+        readonly ManagedIdentitySettings? managedIdentitySettings;
+
+        public SqlConnectionFactory(string connectionString, ManagedIdentitySettings? managedIdentitySettings = null)
+        {
+            this.connectionString = connectionString;
+            this.managedIdentitySettings = managedIdentitySettings;
+        }
+
+        public async Task<SqlConnection> CreateConnection()
```

Comment:
Can we rename this to `CreateConnectionAsync()`? It's not a big deal, but we try to follow this naming convention for methods that return tasks (which is the case now that we may fetch an Azure AD token).

Reply:
Sure. I guess I forgot about this naming pattern :)